src/base/QXmppStanza.h

```diff
@@ -63,6 +66,28 @@ class QXMPP_EXPORT QXmppExtendedAddress
     QSharedDataPointer<QXmppExtendedAddressPrivate> d;
 };
 
+class QXmppE2eeMetadataPrivate;
+
+class QXMPP_EXPORT QXmppE2eeMetadata
+{
+public:
+    QXmppE2eeMetadata();
+    QXmppE2eeMetadata(const QXmppE2eeMetadata &other);
+    ~QXmppE2eeMetadata();
+
+    QXmppE2eeMetadata &operator=(const QXmppE2eeMetadata &other);
+
+    QByteArray senderKey() const;
+    void setSenderKey(const QByteArray &keyId);
+
+    // XEP-0420: Stanza Content Encryption
+    std::optional<QDateTime> sceTimestamp() const;
+    void setSceTimestamp(const std::optional<QDateTime> &timestamp);
```

Comment:
QDateTime itself is already nullable. I think that's already enough. I wouldn't use `std::optional<QString>` either.